arho_feature_template/gui/dialogs/plan_feature_form.py

```diff
@@ -186,6 +186,23 @@ def _format_names(names, last_item_conjucation: str = "ja") -> str:
 
         return True
 
+    def _check_multiple_regulation_groups_with_principal_intended_use_regulations(self) -> bool:
+        if (
+            sum(
+                1
+                for regulation_group_widget in self.regulation_group_widgets
+                for regulation_widget in regulation_group_widget.regulation_widgets
+                for additional_information_widget in regulation_widget.additional_information_widgets
+                if additional_information_widget.config.additional_information_type == "paakayttotarkoitus"
+            )
+            > 1
+        ):
```

Comment:
Selkeämpi ja tehokkaampi on käyttää `any()` funktiota, joka palauttaa Truen heti kun iteraattorista löytyy yksikin kohde joka on tosi.
```suggestion
        if any(
            additional_information_widget.config.additional_information_type == "paakayttotarkoitus"
            for regulation_group_widget in self.regulation_group_widgets
            for regulation_widget in regulation_group_widget.regulation_widgets
            for additional_information_widget in regulation_widget.additional_information_widgets
        ):
```